### ISSUE TITLE
Cannot read property 'kind' of undefined error is shown when deselecting some text

### ISSUE THREAD

**Issue by darind** [Microsoft/monaco-editor#621]
monaco-editor version: 0.10.1
Browser: Google Chrome
OS: Windows 10

1. Open the Monaco Playground: https://microsoft.github.io/monaco-editor/playground.html
2. Add some comment block to the default javascript code so that the final model looks like this:

```javascript
function hello() {
  alert('Hello world!');
}
/*
NativeScript adheres to the CommonJS specification for dealing with
JavaScript modules. The CommonJS require() function is how you import
JavaScript modules defined in other files.
*/
```

3. Select a portion of the text as shown in the attached video and click outside to deselect it

Actual result: The text is deselected but the following error is printed to the browser console:

```
Uncaught Error: Cannot read property 'kind' of undefined

TypeError: Cannot read property 'kind' of undefined
    at Object._t [as isJsxOpeningElement] (typescriptServices.js:33)
    at Object.r [as getDocumentHighlights] (typescriptServices.js:59)
    at C (typescriptServices.js:65)
    at D (typescriptServices.js:65)
    at Object.x [as getOccurrencesAtPosition] (typescriptServices.js:65)
    at e.getOccurrencesAtPosition (worker.js:9)
    at t.e.fmr (workerMain.js:7)
    at t._handleMessage (workerMain.js:7)
    at Object.handleMessage (workerMain.js:7)
    at e._handleMessage (workerMain.js:7)
    at Object._t [as isJsxOpeningElement] (typescriptServices.js:33)
    at Object.r [as getDocumentHighlights] (typescriptServices.js:59)
    at C (typescriptServices.js:65)
    at D (typescriptServices.js:65)
    at Object.x [as getOccurrencesAtPosition] (typescriptServices.js:65)
    at e.getOccurrencesAtPosition (worker.js:9)
    at t.e.fmr (workerMain.js:7)
    at t._handleMessage (workerMain.js:7)
    at Object.handleMessage (workerMain.js:7)
    at e._handleMessage (workerMain.js:7)
    at editor.main.js:31
```

Expected result: The text is deselected without errors

Repro demo: 
![error_repro](https://user-images.githubusercontent.com/270695/32275932-ca0453de-bf15-11e7-9940-22e88dce740e.gif)

P.S. The same error also happens in TypeScript files.

**Comment by canaanjin** [comment]
I also encountered such a problem.
#615

**Comment by luminaxster** [comment]
If you have an overlay over the editor and come back to the editor, at some point it throws the same error.
![image](https://user-images.githubusercontent.com/6577822/35412937-e1b7b6e2-01eb-11e8-89c8-fe3c5d01338b.png)

**Comment by alexandrudima** [comment]
Upstream issue - https://github.com/Microsoft/TypeScript/issues/17245

Fixed by adopting TS 2.7.2 -- https://github.com/Microsoft/monaco-typescript/commit/4514e58f07deefea088fcd44beb4be685135c6fa

**Comment by alexandrudima** [comment]
@jagenjo What version of the editor are you using?
